### PR TITLE
Overhaul grains, take II

### PR DESCRIPTION
**Pull request by C-Lunn**
# Details
Updated grains to use a more Pythonic instantiation method. Hopefully this time it won't break everything!

Tested against mediaapi, squirrel, bobcat, weasel-python-bindings.

# Pivotal Story
Story URL: https://www.pivotaltracker.com/story/show/181357992
https://www.pivotaltracker.com/story/show/173058691

# Related PRs
_Where appropriate. Indicate order to be merged._

# Links to external test runs/working deployment
_Where appropriate, if separate to default CI run_

# Submitter PR Checks
_(tick as appropriate)_

- [x] Added bbc/rd-apmm-cloudfit team as a reviewer
- [x] PR completes task/fixes bug
- [x] Tests exercise code appropriately
- [x] New features and API breaks are flagged in commit messages using magic strings
- [ ] Repo maintainer is notified that a release is required
- [x] Documentation updated (README, Confluence, Docstrings, API spec, Engineering Guide, etc.)
- [ ] Downstream repos have been checked for potential breaks & fixed as needed
- [x] APIs/UIs/CLIs updated as required
- [ ] PR added to Pivotal story
- [ ] Follow-up stories added to Pivotal
- [ ] Any temporary code/configuration removed (e.g. test deployment environment, temporary commontooling branch)
- [ ] Any pins against pre-releases have been removed

# Reviewer PR Checks
_(tick as appropriate)_

- [ ] PR completes task/fixes bug
- [ ] Tests exercise code appropriately
- [ ] Design makes sense, and fits with our current code base
- [ ] Code is easy to follow
- [ ] PR size is sensible
- [ ] Commit history is sensible and tidy

# Info on Cloudfit PRs
- The checks above are guidelines. They don't all have to be ticked, but they should all have been considered.
- For more details on how to asses PRs, see: https://github.com/bbc/rd-apmm-docs-ways-of-working/blob/master/workflow/PRChecklist.md

### CODE REVIEW

File: mediagrains/numpy/numpy_grains/VideoGrain.py [hunk 1]
```diff
@@ -222,6 +222,7 @@ class VideoGrain (bytesgrain.VideoGrain):
     _grain_conversions: Dict[Tuple[CogFrameFormat, CogFrameFormat], ConversionFunc] = {}
 
     def __init__(self,
+                 *args,
```
Comment (j616) [inline mediagrains/numpy/numpy_grains/VideoGrain.py:225]: NIT: Just want to double check this isn't an unintended breaking change. Do the other arguments still work as positional with this in place?

File: mediagrains/grain.py [hunk 2]
```diff
@@ -36,12 +36,12 @@
...
-    from .grains import Grain
+    from .grains import GrainFactory
 
     src_id = uuid1()
     flow_id = uuid5(src_id, "flow_id:test_flow")
 
-    grain1 = Grain(src_id=src_id, flow_id=flow_id)
-    grain2 = Grain(grain1.meta)
+    grain1 = GrainFactory(src_id=src_id, flow_id=flow_id)
+    # grain2 = Grain(grain1.meta)
```
Comment (j616) [inline mediagrains/grain.py:45]: NIT: Are these commented lines intentional? If possible, I'd either re-instate them or remove them entirely. If they need to stay, could you add a comment explaining whats going on.

**Review by j616 — APPROVED** [review]
Couple of NITs but lgtm

File: mediagrains/numpy/numpy_grains/VideoGrain.py (hunk 1 above)
Comment (C-Lunn) [reply mediagrains/numpy/numpy_grains/VideoGrain.py:225]: They don't, but that's fine, as the 'old style' will call it from the `videograin.py` file which _does_ support positional arguments.

**Pull request merged**